packages/edit-post/src/components/editor-regions/index.js

```diff
@@ -8,8 +8,73 @@ import classnames from 'classnames';
  */
 import { navigateRegions } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useSimulatedMediaQuery } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
 
 function EditorRegions( { footer, header, sidebar, content, publish, className } ) {
+	const resizableStylesheets = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getSettings().resizableStylesheets;
+	}, [] );
+
+	const deviceType = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).deviceType();
+	} );
+
+	const [ actualWidth, updateActualWidth ] = useState( window.innerWidth );
+
+	window.addEventListener( 'resize', () => updateActualWidth( window.innerWidth ) );
+
+	const canvasWidth = ( device ) => {
+		let deviceWidth = 0;
+
+		switch ( device ) {
+			case 'Tablet':
```

Comment:
Related to comment above about storing translated text in state: these `switch` statements won't work if the user's language is not set to English.